### PR TITLE
Slmetzler42 Interceptor Controller Subsystem

### PR DESCRIPTION
**Pull request by slmetzler42**

### CODE REVIEW

File: Documentation/Signoffs/IntercepterControllerSubSystem.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Interceptor Controller Subsystem
+
+## **Function:**
+![Elaboration Photo](../Images/Controllers/InterceptorController.png)
+
+Figure 1: Interceptor Controller Subsystem
+
+The goal of this subsystem is to take the outputs from the processer and convert them into inputs that can be used for the mechanical system.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IntercepterControllerSubSystem.md]: I don't understand what this system is doing. Why not just have the processor output something that the firing system can work with?

**Review by JesseTNRoberts — COMMENTED** [review]
As I mentioned in my previous comments, I don't understand the function of this subsystem. Takes output and makes it in to input is not sufficient.

File: Documentation/Signoffs/IntercepterControllerSubSystem.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+The goal of this subsystem is to take the analog outputs produced by the processor and convert them into digital logic that can be used by the mechanical system to move the mechanism , this is due to it only being able to process digital inputs. 
+
+## **Constraints:**
+
+|No.|Constraint|Origin|
+|--|------|-------|
+|1|The Interceptor Controller shall move firing mechanism to 1 of 30 locations |Conceptual Design|
+|2|Must send back signal when aiming is finished to initiate firing | Conceptual Design |
+|3|Must turn initiate firing| Conceptual Design
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IntercepterControllerSubSystem.md]: ??

File: Documentation/Signoffs/IntercepterControllerSubSystem.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+|4|Arduino must be powered with a 6 to 20V voltage source | Device Constraint|
+|5|Arduino's maximum current per I/O pin is 20mA| Device Constraint|
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IntercepterControllerSubSystem.md]: How is this a constraint on the system?

File: Documentation/Signoffs/IntercepterControllerSubSystem.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+|6|Arduino's maximum current per +3.3V pin is 50mA| Device Constraint|
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IntercepterControllerSubSystem.md]: Not a constraint on the system.

File: Documentation/Signoffs/IntercepterControllerSubSystem.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+|7|Maximum voltage for the Stepper Motor 4 to 92V| Device Constraint|
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IntercepterControllerSubSystem.md]: What stepper?

**Review by JesseTNRoberts — COMMENTED** [review]
There are some important constraints here. But I think most of the real constraints aren't well considered. Further, the function of the system is said to make digital signals into analog. If that's the case, why is there a microcontroller here (just use a D2A)? 

The name of the system is controller, yet the function of the system is not described as controlling anything and there are no constraints relevant for a control system.

**Review by JesseTNRoberts — COMMENTED** [review]
This does not address the comments I left. Don't request a review without addressing all comments appropriately.

File: Documentation/Signoffs/IntercepterControllerSubSystem.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
+# Interceptor Controller Subsystem
+
+## **Function:**
+![Elaboration Photo](../Images/Controllers/InterceptorController.png)
+
+Figure 1: Interceptor Controller Subsystem
+
+The goal of this subsystem is to take the analog outputs produced by the processor and the Arduino converts them the into digital outputs that can be used by the mechanical system to move the firing mechanism. The Arduino and Stepper Motor Driver will allow the specific motors the Mechanical Engineering team has picked to change speed, direction to one of the 30 set positions, and drop the ammunition when the firing sequence is activated .
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IntercepterControllerSubSystem.md]: This is unclear. What task is this actually doing?

File: Documentation/Signoffs/IntercepterControllerSubSystem.md
Comment (micahr234) [inline Documentation/Signoffs/IntercepterControllerSubSystem.md:1]: How is the Arduino going to fire the interceptor ball? Is this done with a actuator?

How is this system going to communicate to the image possessing subsystem?

What makes you think that 1.6sec is enough time to aim?

**Review by micahr234 — COMMENTED** [review]
See my comment in the signoff.

Also do you have an extra file in the main directory? It is named "InterceptorControllerSubsystem.md" and it looks outdated.

File: Documentation/Signoffs/IntercepterControllerSubSystem.md
Comment (slmetzler42) [reply Documentation/Signoffs/IntercepterControllerSubSystem.md:1]: Yes, I accidentally made one there when I was trying to create a new subsystem. This is the up to date one.

Comment (slmetzler42) [reply Documentation/Signoffs/IntercepterControllerSubSystem.md:1]: I also reached out to the ME team and as of right now they have no information on the firing mechanism. We are trying to get them to figure out what they are doing before our meeting time with them, but I'm not sure how much they will get done by the 9/4. They are still figuring out what we will be firing at the incoming golf balls, the magazine to hold the ammo, and so they haven't picked a device to drop the ammunition into the motor for firing.

File: Documentation/Signoffs/IntercepterControllerSubSystem.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
...e to pair with the stepper motor driver and with a large data base of research, experiments, and recommendations. This also allows the team to alleviate some of the coding within the processor making it easier to trouble shoot problems, breaks up coding assignments, and allows for specific testing with the ME team to fine tune the speed and positions.The arduino will need to set the motors to move at 200 RPM [9] to allow ample time for firing. The worse possible case, with it needing to fire at the first or last line, it can take anywhere between 340 and 577 depending on the diameter of the position mechanism. The motor used for the firing mechanism runs at 10,000 RMPs and will be plenty fast enough to hit the object in time. The arduino will also communicate with the The Jetson Nano which will communicate with the Arduino using SPI interface. The pins: 'spi_ce0_n' (Enable), 'spi_clk' (Clock), 'spi_mosi' (Master Out Slave In), and 'spi_miso' (Master In Slave Out) the pins will handle data transmission from the Jetson Nano to the Arduino and vice versa. By connecting these pins it will be able to send location information and firing information while receiving data from the Arduino.
```
Comment (micahr234) [inline Documentation/Signoffs/IntercepterControllerSubSystem.md]: "between 340 and 577" where are the units?

How do you know that 200RPM will allow ample time for firing?

File: Documentation/Signoffs/IntercepterControllerSubSystem.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+2.	After completing the movement to the desired position, and the projectile is within set "kill zone", the firing mechanism is engaged firing only one shot.
+3.	Using the 5V pin on the processor to power the adruino.
+4.  To be able to move to the correct coordinates the motors need to move in either direction along the x and y axis.
+5.  Per Devcom rule books the projectile must be fired at a safe speed and stay within a distance of 6ft *this speed will be found by the ME team*
+6.  Based off calculations made with the dimensions of the game board it will take the projectile approximately 1.95 seconds to 7.4 seconds to travel down the line given the two different height points. However, after subtracting the worst-case scenario with sensing, processing, and travel time this leaves 1.6 seconds for the firing mechanism to move into position and fire at the object before collision.
+
+
+## **Buildable Schematics:**
+
+![Elaboration Photo](../Images/Controllers/Schematic3.png)
```
Comment (micahr234) [inline Documentation/Signoffs/IntercepterControllerSubSystem.md]: Where is SPI connection to the processor?

File: Documentation/Signoffs/IntercepterControllerSubSystem.md
Comment (micahr234) [reply Documentation/Signoffs/IntercepterControllerSubSystem.md:1]: Remove outdated file.

You need to add a paragraph about all the information you know about the firing mechanism. You need to at least have a plan maybe not all the details before signoff.

**Review by micahr234 — APPROVED** [review]
I am approving even though the ME team has not finished. Changes will have to be made to accommodate their design in the future.

**Pull request merged**